### ISSUE TITLE
Bug in store then sell object

### ISSUE THREAD

**Issue by Meklon2007** [relarn/relarn#3]
When i tried to sell the dagger with -2, it turned out that the store buys it for 550000, which is clearly a lot.

It looks like an error in object.c:
```
for (izarg = obj.iarg; izarg; izarg--) {
            if (value >= 500000L) break;
            value = 14 * (67+value)/10;
        }/* for */
```

In Ulan, the calculation was not performed with a negative value:

```
while ((izarg-- > 0) && ((value=14*(67+value)/10) < 500000L))
	;
```

**Comment by suetanvil** [comment]
That's definitely the problem.  The `for` condition is wrong.   It should be `for (izarg = obj.iarg; izarg > 0; izarg--) {`.

My guess is that at some point, I made the `iarg` field unsigned and then made it signed again sometime later once I realized what a headache that was.

**Comment by suetanvil** [comment]
This is now fixed, as of release 2.2.2.  Thanks for the report.
